### PR TITLE
Feature/retrieve and consolidate snapshots 127815131

### PR DESCRIPTION
**Pull request by srgoldman**
Get the comparison results from Percy for the latest build and generate a pdf file for compliance to sign and keep.

### CODE REVIEW

File: lib/flash_flow/release/percy_client.rb [hunk 1]
```diff
@@ -1,34 +1,56 @@
...
       end
 
       private
 
-      def initialize_connection!(token)
-        if token.nil?
-          raise RuntimeError.new("Percy token must be set in your flash flow config.")
+      def get_build_id(sha=nil)
+        build = find_latest_by_sha(sha || get_latest_sha)
+        build['web-url'].split('/').last
```
Comment (50kudos) [inline lib/flash_flow/release/percy_client.rb:39]: Ahh sorry Seth, I can expose build_id to you directly.

Comment (50kudos) [inline lib/flash_flow/release/percy_client.rb:39]: Basically just remove the last line of `find_build_by_commit_id` method, and you will have the build id

Comment (srgoldman) [inline lib/flash_flow/release/percy_client.rb:39]: I don't understand your comment. find_build_by_commit_id is a chain of method calls.

Comment (50kudos) [inline lib/flash_flow/release/percy_client.rb:39]: I mean I filtered only `attributes` keys in the `find_build_by_commit_id` method.
`find_latest_by_sha` can give `build['id']` to you if you want. But getting it from `build['web-url']` is also ok.

File: bin/flash_flow [hunk 2]
```diff
@@ -30,6 +32,8 @@ case
     FlashFlow::Merge::Status.new(FlashFlow::Config.configuration.issue_tracker, FlashFlow::Config.configuration.branches, FlashFlow::Config.configuration.branch_info_file, FlashFlow::Config.configuration.git, logger: FlashFlow::Config.configuration.logger).status_html
   when options[:release_branches]
     FlashFlow::Merge::Master.new(options).run
+  when options[:gen_pdf_diffs]
+    FlashFlow::Release::Base.new(FlashFlow::Config.configuration.release).gen_pdf_diffs(*options[:gen_pdf_diffs])
```
Comment (50kudos) [inline bin/flash_flow:36]: I think you do not need the splat operator (*) here

File: flash_flow.yml.erb.example [hunk 3]
```diff
@@ -85,7 +85,24 @@ branch_info_file: 'your/random/file'
...
+#    name: 'FlashFlow::Release::PercyClient'
 #    token: <%= ENV['PERCY_TOKEN'] %>
 #    repo: # Your github repo. For example, the flash_flow repo is 'flashfunders/flash_flow'
 #    api_url: 'https://percy.io/api/v1/'
+#    email: 'foo@bar.com'
+#    cc_email: 'bar@foo.com'
+#smtp:
+#  settings:
+#    domain: 'yourdomain.com'
+#    address: <%= ENV['SMTP_ADDRESS'] %>
```
Comment (50kudos) [inline flash_flow.yml.erb.example:97]: My bad, this is already changed in flashfunders config file, to be `SMTP_HOST`

File: lib/flash_flow/data/base.rb [hunk 4]
```diff
@@ -28,13 +29,16 @@ def initialize_collection(branch_config)
           collection = Collection.fetch(branch_config)
           # Order matters. We are marking the PRs as current, not the branches stored in the json
           collection.mark_all_as_current
-          collection.reverse_merge(stored_collection)
+          collection = collection.reverse_merge(stored_collection)
```
Comment (50kudos) [inline lib/flash_flow/data/base.rb]: Aha interesting, that was a bug?

File: lib/flash_flow/data/base.rb [hunk 5]
```diff
@@ -28,13 +29,16 @@ def initialize_collection(branch_config)
...
 
         else
           collection = stored_collection
           collection.mark_all_as_current
-          collection
         end
+
+        collection.branches.delete_if { |k, v|  TimeHelper.massage_time(v.updated_at) < Time.now - TimeHelper.two_weeks  }
+
+        collection
```
Comment (50kudos) [inline lib/flash_flow/data/base.rb:41]: I think the line is not needed.

**Comment by 50kudos** [comment]
In `test/lib/release/test_pdf_diff_generator.rb`, you pass in the entire json response, should it be a separate fixture file?

**Comment by 50kudos** [comment]
Some comments collapsed by rebasing. So you might want to expand those to consider but those no show stoppers. 🍡

**Pull request merged**